File: Python/6py 2/my_first_interpreter.py
"""
|--------------------------------------------------------------------------
| Esolang Interpreters #1 - Introduction to Esolangs and My First Interpreter (MiniStringFuck)
|--------------------------------------------------------------------------
|
"""

def my_first_interpreter(code):
   res, x = "", 0
   for char in code:
      if char == "+":
         x = (x + 1) % 256
      elif char == ".":
         res += chr(x)
   return res


def my_first_interpreter(code):
   return "".join([chr(code[:i].count("+") % 256) for i, c in enumerate(code) if c == "."])


def my_first_interpreter(code):
   out = ''
   al = 0
   for c in code:
      if c == '+' : al = (al+1)%256
      if c == '.' : out += (chr(al))
   return out


print(my_first_interpreter('+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.'), 'ABCDEFGHIJKLMNOPQRSTUVWXYZ')

File: PHP/6php 2/my_first_interpreter.php
<?php
/*
|--------------------------------------------------------------------------
| Esolang Interpreters #1 - Introduction to Esolangs and My First Interpreter (MiniStringFuck)
|--------------------------------------------------------------------------
|
*/

function my_first_interpreter($code) {
   $res = ""; $x = 0;
   for ($i = 0; $i < strlen($code); $i++) {
      if ($code[$i] === "+") {
         $x = ($x + 1) % 256;
      } elseif ($code[$i] === ".") {
         $res .= chr($x);
      }
   }
   return $res;
}


function my_first_interpreter(string $code): string {
   $cell = 0; $res = '';
   for ($i = 0; $i < strlen($code); $i++)
      switch ($code[$i]) {
         case '+': $cell++; break;
         case '.': $res .= chr($cell); break;
         default: break;
      }
   return $res;
}


function my_first_interpreter(string $code): string {
   $cell = 0;
   $res = '';
   for ($i = 0; $i < strlen($code); $i++)
      switch ($code[$i]) {
         case '+': $cell++; break;
         case '.': $res .= chr($cell); break;
         default: break;
      }
   return $res;
}


print_r(my_first_interpreter('+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.+.') . ' ABCDEFGHIJKLMNOPQRSTUVWXYZ');